File: AdvantageFramework.Core/Database/ConnectionDatabaseProfile.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AdvantageFramework.Core.Database
{
    public partial class ConnectionDatabaseProfile //: AdvantageFramework.Core.BaseClasses.BaseClass
    {

        #region  Constants 



        #endregion

        #region  Enum 

        public enum Properties
        {
            ServerName,
            DatabaseName,
            UserName,
            Password
        }

        #endregion

        #region  Variables 

        private string _DecryptedPassword = string.Empty;

        #endregion

        #region  Properties 

        //[System.Xml.Serialization.XmlIgnore()]
        //[AdvantageFramework.BaseClasses.Attributes.Entity(ShowColumnInGrid = false)]
        //public Guid GUID { get; set; }
        //[AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired = true, DisplayFormat = "")]
        //public string ServerName { get; set; }
        //[AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired = true, DisplayFormat = "")]
        //public string DatabaseName { get; set; }
        //[AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired = true, DisplayFormat = "")]
        //public string UserName { get; set; }
        //[AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired = true, DisplayFormat = "")]
        //public string Password { get; set; }

        #endregion

        #region  Methods 

        //public ConnectionDatabaseProfile()
        //{
        //    GUID = Guid.NewGuid();
        //}

        //public ConnectionDatabaseProfile Copy()
        //{
        //    ConnectionDatabaseProfile CopyRet = default;

        //    // objects
        //    ConnectionDatabaseProfile ConnectionDatabaseProfile = null;
        //    ConnectionDatabaseProfile = new ConnectionDatabaseProfile();
        //    ConnectionDatabaseProfile.ServerName = ServerName;
        //    ConnectionDatabaseProfile.DatabaseName = DatabaseName;
        //    ConnectionDatabaseProfile.UserName = UserName;
        //    ConnectionDatabaseProfile.Password = Password;
        //    CopyRet = ConnectionDatabaseProfile;
        //    return CopyRet;
        //}

        //public string GetDecryptPassword()
        //{
        //    string GetDecryptPasswordRet = default;
        //    string DecryptedPassword = string.Empty;
        //    try
        //    {
        //        DecryptedPassword = AdvantageFramework.Core.Security.Encryption.Decrypt(Password);
        //    }
        //    catch (Exception ex)
        //    {
        //        DecryptedPassword = string.Empty;
        //    }

        //    GetDecryptPasswordRet = DecryptedPassword;
        //    return GetDecryptPasswordRet;
        //}

        //public string GetConnectionString(AdvantageFramework.Core.Security.Application Application)
        //{
        //    string GetConnectionStringRet = default;
        //    string ConnectionString = string.Empty;
        //    ConnectionString = AdvantageFramework.Core.Database.CreateConnectionString(ServerName, DatabaseName, false, UserName, GetDecryptPassword(), Application.ToString);
        //    GetConnectionStringRet = ConnectionString;
        //    return GetConnectionStringRet;
        //}

        //public override string ToString()
        //{
        //    string ToStringRet = default;
        //    ToStringRet = AdvantageFramework.Core.StringUtilities.Methods.AppendTrailingCharacter(ServerName, @"\") + DatabaseName;
        //    return ToStringRet;
        //}

        //public void SetError(string ErrorText)
        //{
        //    string FullErrorText = string.Empty;
        //    FullErrorText = _EntityError;
        //    if (string.IsNullOrWhiteSpace(ErrorText))
        //    {

        //        // _ErrorHashtable("ServerName") = String.Empty
        //        _ErrorHashtable("DatabaseName") = string.Empty;
        //    }
        //    else
        //    {

        //        // _ErrorHashtable("ServerName") = ErrorText
        //        _ErrorHashtable("DatabaseName") = ErrorText;
        //    }

        //    _EntityError = Interaction.IIf(string.IsNullOrEmpty(FullErrorText), ErrorText, FullErrorText + Environment.NewLine + ErrorText);
        //}

        #endregion

    }
}

File: AdvantageFramework/Database/ConnectionDatabaseProfile.vb
Namespace Database

    <Serializable()>
    Public Class ConnectionDatabaseProfile
        Inherits AdvantageFramework.BaseClasses.BaseClass

#Region " Constants "



#End Region

#Region " Enum "

        Public Enum Properties
            ServerName
            DatabaseName
            UserName
            Password
        End Enum

#End Region

#Region " Variables "

        Private _DecryptedPassword As String = String.Empty

#End Region

#Region " Properties "

        <Xml.Serialization.XmlIgnore()>
        <AdvantageFramework.BaseClasses.Attributes.Entity(ShowColumnInGrid:=False)>
        Public Property GUID As System.Guid
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=True, DisplayFormat:="")>
        Public Property ServerName() As String
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=True, DisplayFormat:="")>
        Public Property DatabaseName() As String
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=True, DisplayFormat:="")>
        Public Property UserName() As String
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=True, DisplayFormat:="")>
        Public Property Password() As String

#End Region

#Region " Methods "

        Public Sub New()

            Me.GUID = System.Guid.NewGuid

        End Sub
        Public Function Copy() As ConnectionDatabaseProfile

            'objects
            Dim ConnectionDatabaseProfile As ConnectionDatabaseProfile = Nothing

            ConnectionDatabaseProfile = New ConnectionDatabaseProfile

            ConnectionDatabaseProfile.ServerName = Me.ServerName
            ConnectionDatabaseProfile.DatabaseName = Me.DatabaseName
            ConnectionDatabaseProfile.UserName = Me.UserName
            ConnectionDatabaseProfile.Password = Me.Password

            Copy = ConnectionDatabaseProfile

        End Function
        Public Function GetDecryptPassword() As String

            Dim DecryptedPassword As String = String.Empty

            Try

                DecryptedPassword = AdvantageFramework.Security.Encryption.Decrypt(Me.Password)

            Catch ex As Exception
                DecryptedPassword = String.Empty
            End Try

            GetDecryptPassword = DecryptedPassword

        End Function
        Public Function GetConnectionString(Application As AdvantageFramework.Security.Application) As String

            Dim ConnectionString As String = String.Empty

            ConnectionString = AdvantageFramework.Database.CreateConnectionString(Me.ServerName, Me.DatabaseName, False, Me.UserName, Me.GetDecryptPassword(), Application.ToString)

            GetConnectionString = ConnectionString

        End Function
        Public Overrides Function ToString() As String

            ToString = AdvantageFramework.StringUtilities.AppendTrailingCharacter(Me.ServerName, "\") & Me.DatabaseName

        End Function
        Public Sub SetError(ErrorText As String)

            Dim FullErrorText As String = String.Empty

            FullErrorText = _EntityError

            If String.IsNullOrWhiteSpace(ErrorText) Then

                '_ErrorHashtable("ServerName") = String.Empty
                _ErrorHashtable("DatabaseName") = String.Empty

            Else

                '_ErrorHashtable("ServerName") = ErrorText
                _ErrorHashtable("DatabaseName") = ErrorText

            End If

            _EntityError = IIf(FullErrorText = "", ErrorText, FullErrorText & Environment.NewLine & ErrorText)

        End Sub

#End Region

    End Class

End Namespace

